client/src/main/kotlin/no/nav/su/se/bakover/client/person/PersonClient.kt

```diff
@@ -19,61 +24,111 @@ internal data class PersonClientConfig(
     val pdlClientConfig: PdlClientConfig,
 )
 
+/**
+ * [CacheKey] sørger for at rettighetene til brukerne blir ivaretatt, mens systembrukeren har tilgang til alt.
+ *
+ * @param personCache Brukes av både brukere og systembrukeren.
+ * @param aktørIdCache Brukes av både brukere og systembrukeren.
+ */
 internal class PersonClient(
-    val config: PersonClientConfig,
+    private val config: PersonClientConfig,
+    private val pdlClient: PdlClient = PdlClient(config.pdlClientConfig),
+    private val hentBrukerToken: () -> JwtToken.BrukerToken = {
+        JwtToken.BrukerToken.fraMdc()
+    },
+    private val personCache: Cache<CacheKey, Person> = newCache(),
+    private val aktørIdCache: Cache<CacheKey, AktørId> = newCache(),
 ) : PersonOppslag {
-    private val pdlClient = PdlClient(config.pdlClientConfig)
 
-    override fun person(fnr: Fnr): Either<KunneIkkeHentePerson, Person> = pdlClient.person(fnr).map { toPerson(it) }
-    override fun personMedSystembruker(fnr: Fnr): Either<KunneIkkeHentePerson, Person> =
-        pdlClient.personForSystembruker(fnr).map { toPerson(it) }
+    private fun <Value, Error> Cache<CacheKey, Value>.getOrAdd(
+        key: CacheKey,
```

Comment:
Antar at cachekey er åpen og det kunne vært en mulighet å laget egne key-typer for de aktuelle tilfellene?

Reply:
Caffeine tar bare inn Any, Any (generics) som key-value og gjøre bare en equals for å se om det cache hits/misses. Dette er vår type. Basert på hvor stor plass denne cachen vil ta, kan vi se om vi kan gjøre optimaliseringer på f.eks. og ha fler caches (fnr->Person) og JWT->fnr)